dash/testing/browser.py

```diff
@@ -247,22 +252,20 @@ def open_new_tab(self, url=None):
         )
 
     def get_webdriver(self, remote):
-        # occasionally the browser fails to start - give it 3 tries
-        for i in reversed(range(3)):
-            try:
-                return (
-                    getattr(self, "_get_{}".format(self._browser))()
-                    if remote is None
-                    else webdriver.Remote(
-                        command_executor=remote,
-                        desired_capabilities=getattr(
-                            DesiredCapabilities, self._browser.upper()
-                        ),
-                    )
+        try:
+            return (
+                getattr(self, "_get_{}".format(self._browser))()
+                if remote is None
+                else webdriver.Remote(
+                    command_executor=remote,
+                    desired_capabilities=getattr(
+                        DesiredCapabilities, self._browser.upper()
+                    ),
                 )
-            except WebDriverException:
-                if not i:
-                    raise
+            )
+        except WebDriverException:
+            logger.exception("<<<Webdriver not initialized correctly>>>")
```

Comment:
@alexcjohnson  but the `logger.exception` will record all the exceptions, so it's not swallowed by this function. It depends on how we configure `pytest`, so all the logs could be accessible in the circleci.

Reply:
true, and this has the advantage that you'll actually see *all* the exceptions - ie if it fails 5 times you'll see all 5 - and I guess the `until` will ensure that at the end of it all there at least *is* an exception. OK, let's do it.